### DISCUSSION TITLE
Feature: "show underlying data" for Prometheus

### DISCUSSION THREAD

**Discussion by bboreham** [grafana/grafana#70385] [Explore]
The standard way to get data for a panel from Prometheus is the "range query" - this is sampled at a fixed rate, e.g. every 1 minute.
But the underlying data may be more or less frequent, or even quite irregular.

It is possible in Grafana to see the underlying data - for a 1-hour range query on `foo` you change the option from "range" to "instant" and change the query to `foo[1h]`.  Now you will get points displayed with their real timestamps, not sampled.

This two-step process is a bit advanced for many users, however.   I would like Grafana Explore to have a "view underlying data" button built in which achieves the same effect.

It wouldn't work for more advanced cases - a join, for instance, can only work when the timestamps line up.  This is a good reason why range queries always sample.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). We are migrating this discussion to an issue and closing the discussion. The issue is #82650. Feel free to continue the discussion around this there. Thank you!
